### PR TITLE
Add more performant sensor spawner

### PR DESCRIPTION
**Pull request by avanbrunt-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code).
- [x] Linter has passed locally and any fixes were made for failures.
- [x] A self-review of the code has been done.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes (not tied to bugs/features)
- [ ] Other (please describe):

## What is the ticket or issue number?

EA-21703

## Pull Request Description

Change spawn worker logic to use more performant sensor fetching and status checking

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Manual

### CODE REVIEW

**Review by kebringer-cb — APPROVED** [review]

**Review by emitreva-cb — APPROVED** [review]

**Comment by emitreva-cb** [comment]
Just wondering whether we need additional unit tests.

**Comment by avanbrunt-cb** [comment]
> Just wondering whether we need additional unit tests.

I would agree however we don't have any unit test framework built and this change is not large enough to warrant the amount of effort needed to build a EDR framework

**Comment by emitreva-cb** [comment]
> > Just wondering whether we need additional unit tests.
> 
> I would agree however we don't have any unit test framework built and this change is not large enough to warrant the amount of effort needed to build a EDR framework

Disregard it :) I thought it is the sdk.

File: src/cbapi/live_response_api.py [hunk 1]
```diff
@@ -985,9 +985,21 @@ def _spawn_new_workers(self):
...
-                   and s.id not in self._job_workers
-                   and s.status == "Online"]
+        sensors = []
+
+        for sensor_id in self._unscheduled_jobs:
+            sensor = self._cb.select(Sensor, sensor_id)
+            if sensor_id in self._job_workers:
+                log.debug("Skipping sensor {} already has a worker created".format(sensor_id))
+            elif sensor.status == "Online":
+                sensors.append()
```
Comment (7a6570) [inline src/cbapi/live_response_api.py]: shouldn't it be sensors.append(sensor) ? sensors.append() does not append anything to the sensors list

Comment (7a6570) [reply src/cbapi/live_response_api.py]: sorry, i'm wrong, this is fixed in the second commit

Comment (avanbrunt-cb) [reply src/cbapi/live_response_api.py]: Thanks! I had forgot that and it crashed on me during my test :)

**Pull request merged**